src/vmm/src/resources.rs

```diff
@@ -171,6 +172,44 @@ impl VmResources {
         Ok(resources)
     }
 
+    /// Updates the resources from a restored device (used for configuring resources when
+    /// restoring from a snapshot).
+    pub fn update_from_restored_device(&mut self, device: SharedDeviceType) {
+        match device {
+            SharedDeviceType::SharedBlock(block) => {
+                self.block.add_device(block);
+            }
+
+            SharedDeviceType::SharedNetwork(network) => {
+                if let Ok(net) = network.lock() {
```

Comment:
let's do an `expect("Poisoned lock")` here, since this is an irrecoverable error

Reply:
I changed it to use the `expect("Poisoned lock")` but doing that introduced a borrow checker issue since `network.lock().expect()` results in a borrow but then `network` gets moved with `self.net_builder.add_device(network);`. A workaround I added was a putting the `if` in its own scope but I feel like there must be a better , more idiomatic Rust way? I tried cloning the `network` device but that also had a separate borrow checker issue.

Reply:
you may call `drop(net)` before calling `self.net_builder.add_device(network);`, but I think the current approach is fine